modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -155,20 +156,37 @@ private String getVersion(Extension extension)
 
     private void declareConfigurations(DeclarationDescriptor declaration, Class<?> extensionType)
     {
-        Configurations configurations = extensionType.getAnnotation(Configurations.class);
-        if (configurations != null)
+        Class<?>[] configurationClasses = getConfigurationClasses(extensionType);
+        if (ArrayUtils.isEmpty(configurationClasses))
         {
-            for (Class<?> declaringClass : configurations.value())
-            {
-                declareConfiguration(declaration, declaringClass);
-            }
+            declareConfiguration(declaration, extensionType);
         }
         else
         {
-            declareConfiguration(declaration, extensionType);
+            for (Class<?> configurationClass : configurationClasses)
+            {
+                //TODO: MULE-9220
+                String msg = "Class %s can't declare a configuration and operations at the same time";
+                Arrays.stream(getOperationClasses(extensionType))
+                        .forEach(operationClass -> {
+                            boolean ConfigurationIsAnOperation = configurationClass.isAssignableFrom(operationClass);
```

Comment:
there seems to be no point in having this boolean variable, you can just put the expression in the if